Add mock functions under MockUefiLib and Create Mock for AcpiTable, FirmwareVolume2 and AcpiSystemDescriptionTable protocol

TsunFeng opened the pull request:
## Description

Add mock functions under MockUefiLib and Create Mock for AcpiTable, FirmwareVolume2 and AcpiSystemDescriptionTable protocol

For details on how to complete to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Unit tests component can call these mock functions success

## Integration Instructions

N/A

makubacki committed:
MdeModulePkg: CodeQL Fixes.
Includes changes across the repo for the following CodeQL rules:
- cpp/comparison-with-wider-type
- cpp/overflow-buffer
- cpp/redundant-null-check-param
- cpp/uselesstest

Co-authored-by: Taylor Beebe <tabeebe@microsoft.com>
Co-authored-by: pohanch <125842322+pohanch@users.noreply.github.com>
Co-authored-by: kenlautner <85201046+kenlautner@users.noreply.github.com>
Co-authored-by: Oliver Smith-Denny <osde@linux.microsoft.com>
Co-authored-by: Sean Brogan <sean.brogan@microsoft.com>
Co-authored-by: Aaron <105021049+apop5@users.noreply.github.com>

TsunFeng committed:
Add mock functions under MockUefiLib and Create Mock for AcpiTable, FirmwareVolume2 and AcpiSystemDescriptionTable protocol

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiLib.h:
Can you please make a commit in this PR for each individual change? For example, MockAcpiTable content in one commit and MockFirmwareVolume2 content in another.

It is fine for the PR to contain all the changes here now, but it would be easier to review and follow in source history if changes that are not related are in separate commits. Then the PR can be completed as "Rebase & FF" instead of "Squash" and preserve teh commits.

makubacki replied:
Also, when you do this ensure your branch is up-to-date and there is not a merge commit in it, so the commits can be merged cleanly to release/202311.

TsunFeng replied:
Thank you for your advice. I understand that creating separate commits for unrelated changes will make the code review process more efficient. I will make sure to follow this practice in the future.

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockAcpiSystemDescriptionTable.h:
```diff
@@ -0,0 +1,115 @@
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_ACPI_SYSTEM_DESCRIPTION_TABLE_H
+#define MOCK_ACPI_SYSTEM_DESCRIPTION_TABLE_H
```
For all header guards add the `_` postfix.

```suggestion
#ifndef MOCK_ACPI_SYSTEM_DESCRIPTION_TABLE_H_
#define MOCK_ACPI_SYSTEM_DESCRIPTION_TABLE_H_
```

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockAcpiSystemDescriptionTable.h:
```diff
@@ -0,0 +1,115 @@
+  extern EFI_ACPI_SDT_PROTOCOL  *gAcpiSdtProtocol;
+}
+
+#endif // ACPI_SYSTEM_DESCRIPTION_TABLE_H
```
```suggestion
#endif // MOCK_ACPI_SYSTEM_DESCRIPTION_TABLE_H_
```

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,121 @@
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FV_ReadFile,
```
Was there a reason to change the naming convention for some of these?

TsunFeng replied:
The build failure is due to a naming conflict between this function and a function with the same name in the unit test kernel library.

makubacki replied:
Can you link to that function please?

TsunFeng replied:
Hi @makubacki,
The function is a windows api:
https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfile
The error message in build log as below:
![image](https://github.com/user-attachments/assets/d6c40936-50d5-4bad-9174-b72d6982b9b6)

thanks,
TF

VivianNK replied:
Hi @TsunFeng, please rename to `FvReadFile` to follow naming conventions (https://tianocore-docs.github.io/edk2-CCodingStandardsSpecification/release-2.20/4_naming_conventions/#41-general-naming-rules). We generally want to use Pascal Case and not underscores.

makubacki replied:
It might also be helpful to capture the conflict in the PR description as well, but the main concern was the naming convention. People tend to copy/paste and use code as an example and we want examples that follow the conventions.

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,121 @@
+  extern EFI_FIRMWARE_VOLUME2_PROTOCOL  *gFirmwareVolume2Protocol;
+}
+
+#endif // FIRMWARE_VOLUME2_H
```
```suggestion
#endif // FIRMWARE_VOLUME2_H_
```

TsunFeng committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockAcpiSystemDescriptionTable.h

Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: TsunFeng <v-tshuang@microsoft.com>

TsunFeng committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockAcpiSystemDescriptionTable.h

Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: TsunFeng <v-tshuang@microsoft.com>

TsunFeng committed:
For all header guards add the postfix.

VivianNK commented on MdePkg/Test/Mock/Library/GoogleTest/Protocol/MockAcpiSystemDescriptionTable.cpp:
```diff
@@ -0,0 +1,36 @@
+/** @file MockAcpiSystemDescriptionTable.cpp
```
Please remove all the protocol cpp files as in this PR: https://github.com/microsoft/mu_basecore/pull/1111/files/17a677c2f628b17a077ee61739cd249208ee0b1a#r1727583559

TsunFeng committed:
Remove all the protocol cpp for new guidelines

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,148 @@
+    );
+};
+
+extern "C" {
```
Please remove this extern as you only need the one on line 144. Same for other files

TsunFeng committed:
Remove redundant extern in H file

dependabot[bot] committed:
pip: update edk2-pytool-extensions requirement from ~=0.27.11 to ~=0.27.12 (#1133)

Updates the requirements on
[edk2-pytool-extensions](https://github.com/tianocore/edk2-pytool-extensions)
to permit the latest version.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yhsu3 committed:
[Cherry-Pick] Support Report Status Code in the UefiPxe driver.

Report PXE error status via Status Code, with this design,
it will be flexible to register a status code handler
via gEfiRscHandlerProtocolGuid to output the customized error code
to other telemetry service.

The subclass code is `EFI_IO_BUS_IP_NETWORK`

Signed-off-by: Ethan Hsu <Eathonhsu@gmail.com>

TsunFeng commented:
Hi @makubacki,
I've replied to your question in the comment.
Could you please take a look when you have a moment?
Thank you.

TF

apop5 committed:
Revert "MdeModulePkg: CodeQL Fixes."

This reverts commit d74b9cf589ffccbf99d1b315c9528c72c44ad8c7.

TsunFeng commented:
Hi @makubacki,
I would appreciate it if you could review this PR at your earliest convenience.
Your feedback is crucial for moving next unit test.
Thank you.

TF

apop5 commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,144 @@
+EFI_FIRMWARE_VOLUME2_PROTOCOL  FIRMWARE_VOLUME2_PROTOCOL_MOCK = {
+  GetVolumeAttributes,    // EFI_FV_GET_ATTRIBUTES    GetVolumeAttributes;
+  SetVolumeAttributes,    // EFI_FV_SET_ATTRIBUTES    SetVolumeAttributes;
+  FV_ReadFile,            // EFI_FV_READ_FILE         ReadFile;
```
```suggestion
  FvReadFile,            // EFI_FV_READ_FILE         ReadFile;

```

apop5 commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,144 @@
+  SetVolumeAttributes,    // EFI_FV_SET_ATTRIBUTES    SetVolumeAttributes;
+  FV_ReadFile,            // EFI_FV_READ_FILE         ReadFile;
+  ReadSection,            // EFI_FV_READ_SECTION      ReadSection;
+  FV_WriteFile,           // EFI_FV_WRITE_FILE        WriteFile;
```
```suggestion
  FvWriteFile,           // EFI_FV_WRITE_FILE        WriteFile;

```

apop5 commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,144 @@
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FV_ReadFile,
```
```suggestion
    FvReadFile,

```

apop5 commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,144 @@
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FV_WriteFile,
```
```suggestion
    FvWriteFile,

```

VivianNK committed:
Apply edk2 naming conventions

Co-authored-by: Aaron <105021049+apop5@users.noreply.github.com>
Signed-off-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,148 @@
+MOCK_INTERFACE_DEFINITION (MockFirmwareVolume2Protocol);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolume2Protocol, GetVolumeAttributes, 2, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolume2Protocol, SetVolumeAttributes, 2, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolume2Protocol, FV_ReadFile, 7, EFIAPI);
```
@VivianNK, did you mean to leave these with the prior naming convention?

VivianNK replied:
No, missed those. And newlines from last commit need to be removed.

VivianNK commented:
@TsunFeng Please change the PR to target branch `dev/202405`. Moving forward we will be using dev branches to make changes to the latest release branch. 
@apop5

VivianNK committed:
[CHERRY-PICK] MdePkg: Bring in mocks from 2311 (#1139)

## Description

Add mocks from 2311

Cherry picked and squashed the following commits:
a5ba951
5aa1be5

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Tested in 202311 CI

## Integration Instructions

N/A

Signed-off-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>
Co-authored-by: v-sbolisetti <v-sbolisetti@microsoft.com>
Co-authored-by: YiTa-AMI <v-yitawu@microsoft.com>

TsunFeng committed:
Fix uncrustify build error

TsunFeng commented:
Hi @VivianNK & @apop5 
The PR has been updated to target dev/202405. 
Please help merge to the main branch,
Thank You!
TF

VivianNK commented:
> Hi @VivianNK & @apop5 The PR has been updated to target dev/202405. Please help merge to the main branch, Thank You! TF

![image](https://github.com/user-attachments/assets/f57da533-8e44-442d-be30-ad9e9494e1a3)
The target branch still shows as `release/202405` but should be `dev/202405`

TsunFeng commented:
> > Hi @VivianNK & @apop5 The PR has been updated to target dev/202405. Please help merge to the main branch, Thank You! TF
> 
> ![image](https://private-user-images.githubusercontent.com/17439709/368686158-f57da533-8e44-442d-be30-ad9e9494e1a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NDI4OTUsIm5iZiI6MTcyNjc0MjU5NSwicGF0aCI6Ii8xNzQzOTcwOS8zNjg2ODYxNTgtZjU3ZGE1MzMtOGU0NC00NDJkLWJlMzAtYWQ5ZTk0OTRlMWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDEwNDMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5NTZjZmQxODQxOTY5YjI0NmRiZDZlNmY4ODY4ZjE5MWJlMWViMjJiZTYzYzNmNDdhZDMwN2U2MjU5YmY3NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vvktbYuItirh3b5IyRNh50wZCMh1ripQNog2YzHS5sY) The target branch still shows as `release/202405` but should be `dev/202405`

Sorry, it my mistake. I have changed to correct target branch.

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,148 @@
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    FvReadFile,
+
```
please remove these extra newlines

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h:
```diff
@@ -0,0 +1,148 @@
+  GetVolumeAttributes,   // EFI_FV_GET_ATTRIBUTES    GetVolumeAttributes;
+  SetVolumeAttributes,   // EFI_FV_SET_ATTRIBUTES    SetVolumeAttributes;
+  FvReadFile,            // EFI_FV_READ_FILE         ReadFile;
+
```
please remove this file's extra newlines as well

TsunFeng replied:
Done

TsunFeng committed:
Remove extra newlines in MockFirmwareVolume2.h

TsunFeng committed:
Merge branch 'dev/202405' into personal/TsunFeng/NuvotonSioAcpiGT

VivianNK committed:
Merge branch 'dev/202405' into personal/TsunFeng/NuvotonSioAcpiGT

Signed-off-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>
